src/backend/drm/compositor/mod.rs

```diff
@@ -733,12 +724,13 @@ pub struct RenderFrameResult<'a, B: Buffer, F: AsRef<framebuffer::Handle>, E> {
 }
 
 impl<'a, B: Buffer, F: AsRef<framebuffer::Handle>, E> RenderFrameResult<'a, B, F, E> {
-    /// Returns whether the frame would block the current thread when queued
-    pub fn would_block(&self) -> bool {
+    /// Returns a sync point, if synchronization with kms submission can't be guaranteed through the available apis.
+    pub fn needs_sync(&self) -> Option<SyncPoint> {
```

Comment:
I am not sure if we really want to return a sync point here. The reason for `would_block` only returning a `bool` was to prepare for multiple sync points in the future when we want to expose one per plane element too. So the idea was to have a quick way to identify if sync is needed with a bool and leave it to downstream to figure out how to wait. If I would have had more time I thought about adding a helper that can be used to wait or insert it into calloop transparently (taking all sync points into consideration in the future).

Reply:
It should be possible to merge fences, so I am not sure, if we need to make this api unnecessarily complicated. Though if the api can't import a fence, we can't block on the fences of a dmabuf in combination of a render-syncpoint, as we can't get them all into the same api... Though maybe we could handle this internally in the SyncPoint and allow it to contain multiple fences in the future?

Reply:
But I am also happy with just returning a bool.

Reply:
imo a simple bool should be enough until we have to think about merging fences. It could still be handy when we find a proper api for multiple fences as a quick check.